flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/utils/TableUtil.java

```diff
@@ -0,0 +1,429 @@
+	 */
+	public static void assertVectorCols(TableSchema tableSchema, String... selectedCols) {
+		if (selectedCols != null && selectedCols.length != 0) {
+			for (String selectedCol : selectedCols) {
+				if (null != selectedCol) {
+					if (!isVector(findColType(tableSchema, selectedCol))) {
+						throw new IllegalArgumentException("col type must be string " + selectedCol);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Return the columns in the table whose types are string.
+	 *
+	 * @param tableSchema TableSchema
+	 * @return String columns.
+	 */
+	public static String[] getStringCols(TableSchema tableSchema) {
+		return getStringCols(tableSchema, null);
+	}
+
+	/**
+	 * Return the columns in the table whose types are string and are not included in the excludeCols.
+	 *
+	 * <p>If <code>excludeCols</code> is null, return all the string columns.
+	 *
+	 * @param tableSchema TableSchema.
+	 * @param excludeCols The columns who are not considered.
+	 * @return string columns.
+	 */
+	public static String[] getStringCols(TableSchema tableSchema, String[] excludeCols) {
+		ArrayList<String> numericCols = new ArrayList<>();
+		List<String> excludeColsList = null == excludeCols ? null : Arrays.asList(excludeCols);
+		String[] inColNames = tableSchema.getFieldNames();
+		TypeInformation<?>[] inColTypes = tableSchema.getFieldTypes();
+
+		for (int i = 0; i < inColNames.length; i++) {
+			if (isString(inColTypes[i])) {
+				if (null == excludeColsList || !excludeColsList.contains(inColNames[i])) {
+					numericCols.add(inColNames[i]);
+				}
+			}
+		}
+
+		return numericCols.toArray(new String[0]);
+	}
+
+	/**
+	 * Return the columns in the table whose types are numeric.
+	 *
+	 * @param tableSchema TableSchema
+	 * @return numeric columns.
+	 */
+	public static String[] getNumericCols(TableSchema tableSchema) {
+		return getNumericCols(tableSchema, null);
+	}
+
+	/**
+	 * Return the columns in the table whose types are numeric and are not included in the excludeCols.
+	 *
+	 * <p>If <code>excludeCols</code> is null, return all the numeric columns.
+	 *
+	 * @param tableSchema TableSchema.
+	 * @param excludeCols the columns who are not considered.
+	 * @return numeric columns.
+	 */
+	public static String[] getNumericCols(TableSchema tableSchema, String[] excludeCols) {
+		ArrayList<String> numericCols = new ArrayList<>();
+		List<String> excludeColsList = (null == excludeCols ? null : Arrays.asList(excludeCols));
+		String[] inColNames = tableSchema.getFieldNames();
+		TypeInformation<?>[] inColTypes = tableSchema.getFieldTypes();
+
+		for (int i = 0; i < inColNames.length; i++) {
+			if (isNumber(inColTypes[i])) {
+				if (null == excludeColsList || !excludeColsList.contains(inColNames[i])) {
+					numericCols.add(inColNames[i]);
+				}
+			}
+		}
+
+		return numericCols.toArray(new String[0]);
+	}
+
+	/**
+	 * Get the columns from featureCols who are included in the <code>categoricalCols</code>, and the columns whose
+	 * types are string or boolean.
```

Comment:
could you elaborate what scenarios would require this API?

Reply:
An example of use this API is https://issues.apache.org/jira/browse/FLINK-13835.

Reply:
can you add a simple example in the javadoc?

Reply:
Thanks, added